lua/conform/formatters/prettierd.lua

```diff
@@ -1,5 +1,66 @@
 local fs = require("conform.fs")
 local util = require("conform.util")
+
+local config_file_names = {
+  -- https://prettier.io/docs/en/configuration.html
+  ".prettierrc",
+  ".prettierrc.json",
+  ".prettierrc.yml",
+  ".prettierrc.yaml",
+  ".prettierrc.json5",
+  ".prettierrc.js",
+  ".prettierrc.cjs",
+  ".prettierrc.mjs",
+  ".prettierrc.toml",
+  "prettier.config.js",
+  "prettier.config.cjs",
+  "prettier.config.mjs",
+}
+
+---@param file string
+---@return nil|table
+local function read_json(file)
+  local f = io.open(file, "r")
+  if not f then
+    error("Unable to open file " .. file)
```

Comment:
So it turns out that's what I meant when I said

> I feel it's more applicable to throw an exception here. I'll check it.

I haven't tested this specific code path cause the `read_json` is an internal util, and we should never reach this line until we don't have bugs in the `package.json` logic.